### PR TITLE
document that Python 3 is to be used (required) now & misc minor

### PR DESCRIPTION
**Pull request by hoijui**

### CODE REVIEW

**Comment by ickc** [comment]
This change should not be merged. People using Python2 can still install panflute. If the dependencies in this repo is properly configured then Python2 users will just have the last panflute which still supports Python2.

**Comment by ickc** [comment]
For Dev install it is still not quite necessary. For properly configured environments (such as one using conda) python is pointing to python3. Although I'm not certain of any example (Arch?), some distro might not have python3, where python is version 3.

**Comment by sergiocorreia** [comment]
I wonder if at this point, should we ever have to mention it? If py2 is unsupported that means that the omen is on the user to find previous versions that work, or to be aware that they have a system with both py2 and py3, and that they should know replace `python` with `python3`?

**Comment by ickc** [comment]
I think if the user install from PyPI and setup.py is setup correctly then it’s automatic to choose the last python2 supported version if they use python2.

However the distro shipped Python might not have Python pointing to python3. So that might be a problem. At this point I’d tend to think it is a distro problem, not yours. The user can simply create a symlink for that.

On the other hand on your side it probably won’t hurt to specifically use python3, until Python4 come out where there’s no plan on that yet.

I tend to think this should be fixed only when there’s real problem some users experienced and should find out what is their config.

**Comment by hoijui** [comment]
As I see it, the end use will most likely want to have the latest version of panflute. The default instructions should make sure that is the case, when he installs. We can not expect this user to know about python 2 and 3 and whether `pip` is python 2 or 3 and so on. I agree that it is not panflutes "fault", but sadly, panflute somehow has to deal with it as the reality it lives in. I guess this PR is not the right way as you say ... do you see any other way?

**Comment by ickc** [comment]
This is a random example I just saw on documenting this: <https://github.com/Nekmo/amazon-dash>.

I would suggest leaving the default as python/pip and mention alone the line of "you may need to use python3/pip3 to ensure you're using the latest version depending on your system."

**Comment by hoijui** [comment]
Good suggestion! :-)

> You may need to use `pip3` explicitly,
> as `pip` still means `pip2` (python 2) on some systems,
> but recent versions of panflute require python 3.

File: README.md [hunk 1]
```diff
@@ -23,14 +23,21 @@ To install panflute, open the command line and type:
 pip install panflute
 ```
 
-Python 3.6+ and PyPy3 are supported (Python 2.7 and Python 3.3-3.5 were supported up to version 1.11.4).
+###### Python version
+
+> You may need to use `pip3`/`python3` explicitly,
+> as `pip`/`python` still means `pip2`/`python2` on some systems.\
+> Python 3.6+ and PyPy3 are currently supported,
+> while Python 2.7 and Python 3.3 - 3.5 were supported up to panflute version 1.11.4.
```
Comment (dhimmel) [inline README.md:31]: why is this section in blockquotes?

also there is a gap in heading level (2 to 6)

**Comment by sergiocorreia** [comment]
Thanks for the PR as well as for the review. At this point I would have thought that py3 is the default so I haven't even thought about making that explicit, but might be good for some to know that they are able to use older versions of panflute that work for py2 (for compat reasons?)

Also, I can wait for edits if you folks want or just merge it as it is.

Thoughts?

**Comment by dhimmel** [comment]
IMO, specifying the minimum supported python is `setup.py` via `python_requires` would be a good idea, i.e. `python_requires='>=3.6'`. This would enforce the minimum version programmatically.

**Comment by ickc** [comment]
See #149. c.f. #150.

**Pull request closed without merging**